pkg/framework/plugins/defaultevictor/defaultevictor.go

```diff
@@ -278,3 +295,41 @@ func getPodIndexerByOwnerRefs(indexName string, handle frameworktypes.Handle) (c
 
 	return indexer, nil
 }
+
+func getNamespacesListByLabelSelector(ctx context.Context, client clientset.Interface, nsLister listersv1.NamespaceLister, labelSelector string) (map[string]struct{}, error) {
+	ret := make(map[string]struct{})
+	namespaceSelector, err := labels.Parse(labelSelector)
+	if err != nil {
+		return ret, err
+	}
+
+	var ns []*v1.Namespace
+	// err is defined above
+	if ns, err = nsLister.List(namespaceSelector); err != nil {
+		return ret, err
+	}
+
+	if len(ns) == 0 {
```

Comment:
This path will get executed every time the selector does not match any namespace. Which will produce a lot of API requests to the apiserver. Something we want to avoid. What case are you addressing by this? All the informers are expected to be synced with the apiserver at this point.

Instead, if a namespace label selector is provided/enabled but the list of matched namespace is zero we should error. So a user can either check whether there's some transient error or to disable the namespace selector or label namespaces or a different action. To avoid cases where the preEviction protection does not work properly and pods are unexpectedly evicted.

Reply:
Good points, thanks. I'll rewrite code soon.